client/src/main/java/io/pravega/client/tables/TableTransaction.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Defines all operations that can be made on a Table Transaction.
+ *
+ * @param <KeyT>   Table Key Type.
+ * @param <ValueT> Table Value Type.
+ */
+public interface TableTransaction<KeyT, ValueT> extends AutoCloseable {
```

Comment:
Please add java doc on what a table transaction is supposed to guarantee. 
How does transaction really work with conditional writes on a per key basis. If multiple keys are written in a transaction that are across different segments, when a transaction is committed, not all keys may correspond to the versions specified in the transaction and their update may fail. What is the semantic for such a transaction?

Reply:
Done.